proposals/ml_fit_class_xgboost.json

```diff
@@ -0,0 +1,155 @@
+                "dimensions": [
+                    {
+                        "type": "geometry"
+                    }
+                ]
+            }
+        },
+        {
+            "name": "learning_rate",
+            "description": "Step size shrinkage used in update to prevent overfitting.",
+            "schema": {
+                "type": "number",
+                "minimum": 0,
+                "default": 0.15
+            }
+        },
+        {
+            "name": "max_depth",
+            "description": "Maximum depth of a tree.",
+            "schema": {
+                "type": "integer",
+                "minimum": 1,
+                "default": 5
+            }
+        },
+        {
+            "name": "min_child_weight",
+            "description": "Minimum sum of instance weight (hessian) needed in a child.",
+            "schema": {
+                "type": "number",
+                "minimum": 0,
+                "default": 1
+            }
+        },
+        {
+            "name": "subsample",
+            "description": "Subsample ratio of the training instance.",
+            "optional": true,
+            "default": 0.8,
+            "schema": {
+                "type": "number",
+                "minimum": 0,
+                "maximum": 1
+            }
+        },
+        {
+            "name": "min_split_loss",
+            "description": "Minimum loss reduction required to make a further partition on a leaf node of the tree.",
+            "optional": true,
+            "default": 1,
+            "schema": {
+                "type": "number",
+                "minimum": 0
+            }
+        },
+        {
+            "name": "max_delta_step",
+            "description": "Maximum delta step we allow each tree's weight estimation to be.",
+            "optional": true,
+            "default": 1,
+            "schema": {
+                "type": "number",
+                "minimum": 0
+            }
+        },
+        {
+            "name": "nfold",
+            "description": "Number of folds for cross-validation.",
+            "optional": true,
+            "default": 5,
+            "schema": {
+                "type": "integer",
+                "minimum": 2
+            }
+        },
+        {
+            "name": "nrounds",
+            "description": "Number of boosting rounds.",
+            "optional": true,
+            "default": 100,
+            "schema": {
+                "type": "integer",
+                "minimum": 1
+            }
+        },
+        {
+            "name": "early_stopping_rounds",
+            "description": "Activates early stopping. Validation metric needs to improve at least once in every early_stopping_rounds round(s) to continue training.",
+            "optional": true,
+            "default": 20,
+            "schema": {
+                "type": "integer",
+                "minimum": 1
+            }
+        },
+        {
+            "name": "seed",
```

Comment:
a lot of parameters are listed here. Are these based on a particular xgboost implementation? And are we sure that they are translatable to other implementation? Otherwise I think it would make sense to drop a couple from this initial spec, and leave some wiggle room for backend implementers

I remember we had this same issue with the random forest process

Reply:
Yeah, I had checked python and R libraries and these params can be specified : https://xgboost.readthedocs.io/en/stable/parameter.html

I can mabye remove 2 : nfolds and nrounds to be dealt with internally. I set default values for most params though.